hazelcast/src/main/java/com/hazelcast/nio/serialization/ByteArrayObjectDataInput.java

```diff
@@ -51,12 +51,23 @@
     }
 
     ByteArrayObjectDataInput(byte[] data, int offset, SerializationService service, ByteOrder byteOrder) {
-        super();
         this.data = data;
         this.size = data != null ? data.length : 0;
         this.service = service;
-        pos = offset;
-        bigEndian = byteOrder == ByteOrder.BIG_ENDIAN;
+        this.pos = offset;
+        this.bigEndian = byteOrder == ByteOrder.BIG_ENDIAN;
+    }
+   @Override
+    public void init(byte[] data, int offset) {
+        this.data = data;
+        this.size = data != null ? data.length : 0;
+        this.pos = offset;
+        this.mark = 0;
+    }
+
+    @Override
+    public void clear() {
+        this.data = null;
```

Comment:
What about zeroing `size` and `pos`? Otherwise methods like `available()` or `position()` can report incorrect values after clear...